src/rust/engine/options/src/args_tests.rs

```diff
@@ -87,14 +87,16 @@ fn test_float() {
     let assert_float =
         |expected: f64, id: OptionId| assert_eq!(expected, args.get_float(&id).unwrap().unwrap());
 
-    assert_float(4_f64, option_id!(-'j', "jobs"));
+    assert_float(4.0, option_id!(-'j', "jobs"));
     assert_float(3.14, option_id!("foo"));
     assert_float(1.137, option_id!("baz", "spam"));
 
     assert!(args.get_float(&option_id!("dne")).unwrap().is_none());
 
     assert_eq!(
-        "Problem parsing --bad value swallow as a float value: invalid float literal".to_owned(),
+        "Problem parsing --bad float value:\n1:swallow\n  ^\n\
```

Comment:
We switch to using the parse.rs methods for everything, including scalars, for uniformity (and to support non-rust literals that python supports, such as underscores in ints and floats)